backend/app/src/main/java/ch/zhaw/studyflow/controllers/CalendarController.java

```diff
@@ -0,0 +1,252 @@
+package ch.zhaw.studyflow.controllers;
+
+import ch.zhaw.studyflow.domain.calendar.Appointment;
+import ch.zhaw.studyflow.domain.calendar.Calendar;
+import ch.zhaw.studyflow.domain.calendar.CalendarManager;
+import ch.zhaw.studyflow.domain.calendar.AppointmentManager;
+import ch.zhaw.studyflow.webserver.annotations.Endpoint;
+import ch.zhaw.studyflow.webserver.annotations.Route;
+import ch.zhaw.studyflow.webserver.http.HttpMethod;
+import ch.zhaw.studyflow.webserver.http.HttpRequest;
+import ch.zhaw.studyflow.webserver.http.HttpResponse;
+import ch.zhaw.studyflow.webserver.http.HttpStatusCode;
+import ch.zhaw.studyflow.webserver.http.contents.JsonContent;
+import ch.zhaw.studyflow.webserver.http.pipeline.RequestContext;
+import ch.zhaw.studyflow.webserver.security.authentication.AuthenticationHandler;
+import ch.zhaw.studyflow.webserver.security.principal.CommonClaims;
+import ch.zhaw.studyflow.webserver.security.principal.Principal;
+import ch.zhaw.studyflow.webserver.security.principal.PrincipalProvider;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+@Route(path = "api/calendar")
+public class CalendarController {
+    private static final Logger logger = Logger.getLogger(CalendarController.class.getName());
+    private final CalendarManager calendarManager;
+    private final AppointmentManager appointmentManager;
+    private final AuthenticationHandler authenticator;
+    private final PrincipalProvider principalProvider;
+
+    public CalendarController(CalendarManager calendarManager, AppointmentManager appointmentManager, AuthenticationHandler authenticator, PrincipalProvider principalProvider) {
+        this.calendarManager = calendarManager;
+        this.appointmentManager = appointmentManager;
+        this.authenticator = authenticator;
+        this.principalProvider = principalProvider;
+    }
+
+    @Route(path = "createCalendar")
```

Comment:
Let's discuss these routs sometime (maybe sunday?).

Reply:
we can do that on monday

Reply:
i would recommend the following routes:

POST "/calendar": creates a new calendar of currently signed in user
GET "/calendar": gets all calendars of currently signed in user

GET "/calendar/:id": a specific calendar; only the assigned user has the authorization to get it
PATCH "/calendar/:id": updates a specific calendar; only the assigned user has the authorization to do it